terraform.sh

```diff
@@ -0,0 +1,191 @@
+  echo "  summ          Generate summary of Terraform plan"
+}
+
+function init_terraform() {
+  if [ -n "$env" ]; then
+    terraform init -reconfigure -backend-config="./env/$env/backend.tfvars"
+  else
+    echo "ERROR: no env configured!"
+    exit 1
+  fi
+}
+
+function list_env() {
+  # Check if env directory exists
+  if [ ! -d "./env" ]; then
+    echo "No environment directory found"
+    exit 1
+  fi
+
+  # List subdirectories under env directory
+  env_list=$(ls -d ./env/*/ 2>/dev/null)
+
+  # Check if there are any subdirectories
+  if [ -z "$env_list" ]; then
+    echo "No environments found"
+    exit 1
+  fi
+
+  # Print the list of environments
+  echo "Available environments:"
+  for env in $env_list; do
+    env_name=$(echo "$env" | sed 's#./env/##;s#/##')
+    echo "- $env_name"
+  done
+}
+
+function other_actions() {
+  if [ -n "$env" ] && [ -n "$action" ]; then
+    terraform $action -var-file="./env/$env/terraform.tfvars" -compact-warnings $other
+  else
+    echo "ERROR: no env or action configured!"
+    exit 1
+  fi
+}
+
+function state_output_taint_actions() {
+  terraform $action $other
+}
+
+function tfsummary() {
+  action="plan"
+  other="-out=tfplan"
+  other_actions
+  if [ -n "$(command -v tf-summarize)" ]; then
+    tf-summarize -separate-tree tfplan && rm tfplan
+  else
+    echo "tf-summarize is not installed"
+  fi
+}
+
+function update_script() {
+  set -x
+  # Check if update is needed
+  if [ -z "$(command -v curl)" ]; then
+    echo "curl not found, cannot update script"
+    exit 1
+  fi
+
+  current_version=$(sed -n '3p' "$0" | awk '{print $3}')
+  repo_url="https://raw.githubusercontent.com/pagopa/eng-common-scripts/main/azure/terraform.sh"
+  response=$(curl -s -w "%{http_code}" "$repo_url")
+
+  if [ "${response: -3}" == "404" ]; then
+    echo "Script non trovato nel repository."
+    exit 1
+  else
+    new_version=$(echo "$response" | sed -n '1p')
+    new_script=$(curl -s "$repo_url")
+  fi
+
+  exit 0
+  ###
+  ### it will be modified in the future releases
+  ###
+  if [ "$(printf '%s\n' "$new_version" "$current_version" | sort -V | head -n 1)" != "$new_version" ]; then
+    # Ask for confirmation
+    read -p "New version $new_version available. Do you want to update? (y/n) " choice
+    case "$choice" in
+      y|Y )
+        echo "Updating script to version $new_version"
+        ;;
+      * )
+        echo "Update canceled"
+        exit 0
+        ;;
+    esac
+  elif [ "$current_version" == "$new_version" ]; then
+    echo "Script is up to date"
+    exit 0
+  fi
+
+
+  # Download new version
+  new_script=$(curl -s "$repo_url")
+  if [ -n "$new_script" ]; then
+    echo "$new_script" > "$0"
+    echo "Script updated to version $new_version"
+  else
+    echo "Failed to download new script"
+    exit 1
+  fi
+}
+
+# Check arguments number
+if [ "$#" -lt 1 ]; then
+  help_usage
+  exit 0
+fi
+
+# Parse arguments
+action=$1
+env=$2
+shift 2
+other=$@
+
+if [ -n "$env" ]; then
```

Comment:
this script is only for azure right? can we move it into azure dir?

Reply:
done